models/iaf_cond_alpha_mc_fixedca.h

```diff
@@ -0,0 +1,681 @@
+#include "archiving_node.h"
+#include "connection.h"
+#include "event.h"
+#include "nest_types.h"
+#include "recordables_map.h"
+#include "ring_buffer.h"
+#include "universal_data_logger.h"
+
+// Includes from sli:
+#include "dictdatum.h"
+#include "name.h"
+
+/* BeginDocumentation
+Name: iaf_cond_alpha_mc_fixedca - PROTOTYPE Multi-compartment conductance-based
+leaky
+integrate-and-fire neuron model with fixed waveform calcium spike.
+
+iaf_cond_alpha_mc_fixedca is an implementation of a multi-compartment spiking
+neuron using IAF dynamics with conductance-based synapses. It serves
+mainly to illustrate the implementation of multicompartment models in
+NEST. It has also a calcium spike at the distal compartment, details of which
+are described in [3].
+
+The model has three compartments: soma, proximal and distal dendrite,
+labeled as s, p, and d, respectively. Compartments are connected through
+passive conductances as follows
+
+C_m.s d/dt V_m.s = ... - g_sp ( V_m.s - V_m.p )
+
+C_m.p d/dt V_m.p = ... - g_sp ( V_m.p - V_m.s ) - g_pd ( V_m.p - V_m.d )
+
+C_m.d d/dt V_m.d = ...                          - g_pd ( V_m.d - V_m.p ) + ICa
+
+A spike is fired when the somatic membrane potential exceeds threshold,
+V_m.s >= V_th. Upon threshold crossing, somatic membrane potential is set to
+a maximum potential, V_m.s == V_max, and somatic leak set to a larger leak value
+for the refractory period. Depending on value of the reset flag, the somatic
+membrane potential maybe reset to reset value at end of refractory period.
+To emulate backpropagating action potential, an alpha shaped current is
+introduced 1ms and 2ms after spike at proximal and distal compartment
+respectively.
+Dendritic membrane potentials are not manipulated after a spike.
+The spike threshold is adaptive, whereby it jumps by a jump value upon crossing,
+and then decays exponentially back to baseline value.
+
+There is one excitatory and one inhibitory conductance-based synapse
+onto each compartment, with alpha-function time course. The alpha
+function is normalised such that an event of weight 1.0 results in a
+peak current of 1 nS at t = tau_syn. Each compartment can also receive
+current input from a current generator, and an external (rheobase)
+current can be set for each compartment.
+
+Synapses, including those for injection external currents, are addressed through
+the receptor types given in the receptor_types entry of the state dictionary.
+Note
+that in contrast to the single-compartment iaf_cond_alpha model, all synaptic
+weights must be positive numbers!
+
+When active flag is true, calcium spike is triggered upon threshold crossing,
+V_m.d >= V_thCa, provided it is not during the calcium refractory period. The
+spike is
+modeled using a fixed waveform, a current array with fixed length.
+The duration of the calcium spike is of same length as its refractory period,
+hence no new calcium spike is triggered while a calcium spike is still active.
```

Comment:
Because of the fixed Ca dynamics, the model can only be used for time step 0.1 ms. This must be documented here, and there must be a test preventing the model from being created when the timestep is different. This should be checked in the copy constructor of the model.